pkg/controller/clusterpool/collections.go

```diff
@@ -229,6 +232,25 @@ func isBroken(cd *hivev1.ClusterDeployment, pool *hivev1.ClusterPool, logger log
 		return true
 	}
 
+	// Check if CD's customization exists in CP invnetory
```

Comment:
This is because of how [pool version is calculated](https://github.com/openshift/hive/blob/master/docs/enhancements/clusterpool-inventory.md#pool-version)
We need to delete unclaimed CD that uses a CDC that is no longer in the inventory, but without using pool version as the indicator. So the CD is not broken, but also not valid anymore. Do you prefer a dedicated function for this?

Reply:
Ahh, I see. Yes, the logic is sound; but we should figure out a way for it to live in/under cdCollection.Stale() instead.

(Also, it looks like I got confused and thought this was checking whether the CDC exists in the database. I think that's because of the variable name `customizationExists`. Perhaps 'existsInInventory` would be clearer?)